### DISCUSSION TITLE
Custom Code Logic on API Submission

### DISCUSSION THREAD

**Discussion by Gokul270797** [directus/directus#11429] [Archive]
Hi I'm new to this directus. 
Creating a simple project on the local system using version 9. 

I have created a collection form with few basic fields. I'm able to perform CRUD operations through API calls. 
Now i need to add some custom logic before inserting the values to the database fields.
Where can i write the custom code for this.

Any simple tutorial available?

Thanks,
Gokul

**Comment by rijkvanzanten** [comment]
Hi Gokul!

You can use a custom hook:

https://docs.directus.io/extensions/hooks/

**Reply by Gokul270797** [reply]
Hi Rijk van Zanten,
Thanks for your reply.
I've seen this custom hooks but what i don't understand is where to implement it.
Can you please suggest few steps for this?

**Reply by rijkvanzanten** [reply]
Have you seen https://docs.directus.io/extensions/creating-extensions/ ?
